stream.go

```diff
@@ -84,6 +86,11 @@ func (s *Stream) Write(p []byte) (n int, err error) {
 	return s.smuxStream.Write(p)
 }
 
+// Reset resets the stream.
+func (s *Stream) Reset() error {
+	return s.smuxStream.Reset()
```

Comment:
This also needs to remove the stream from the swarm.

Actually, close probably shouldn't be removing the stream from the swarm because we now support half-open connections. However, I can't find anything that handles remote closes...

This is going to need some discussion.

(It looks like there's actually a lot more work here. For example, the stream handlers set in `Swarm.Close()` should probably reset streams, not close them, etc.)